lib/spack/spack/test/stage.py

```diff
@@ -879,34 +828,20 @@ def test_diystage_preserve_file(self, tmpdir):
             _file.read() == _readme_contents
 
 
-@pytest.fixture
-def tmp_build_stage_nondir(tmpdir):
-    """Establish the temporary build_stage pointing to non-directory."""
-    test_stage_path = tmpdir.join('stage', 'afile')
-    test_stage_path.ensure(dir=False)
-
-    # Set test_stage_path as the default directory to use for test stages.
-    current = spack.config.get('config:build_stage')
-    stage_dir = os.path.dirname(str(test_stage_path))
-    spack.config.set('config', {'build_stage': [stage_dir]}, scope='user')
-
-    yield test_stage_path
-
-    spack.config.set('config', {'build_stage': current}, scope='user')
-
-
+@pytest.mark.nomockstage
```

Comment:
This doesn't appear to be necessary (or at least the test passes without it)

Reply:
Again, it's a file system "efficiency" issue (i.e., the test doesn't require `mock_stage` so why unnecessarily hit the file system?).

Reply:
Since the `mock_stage` fixture is being called for every test, I don't think undoing it selectively for a few tests will significantly improve the run time. My preference is that since it adds complexity that it is only used where it is needed for a test to pass.

Reply:
Removed in latest commit.